### ISSUE TITLE
Daemon Severely Misbehaving on Arch Linux

### ISSUE THREAD

**Issue by willfindlay** [pop-os/system76-power#268]
**Distribution (run `cat /etc/os-release`):**
```bash
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```
 
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

As reported by pacman -Qs system76-power:
```
local/system76-power 1.1.17-1
    System76 Power Management
```
Other system76 packages installed:
```
local/system76-acpi-dkms 1.0.2+1+g0bc966c-1
    This provides the system76_acpi in-tree driver for systems missing it.
local/system76-dkms 1.0.13-1
    On newer System76 laptops, this driver controls some of the hotkeys and allows for custom fan control.
local/system76-driver 20.04.41-2
    Universal driver for System76 computers
local/system76-firmware 1.0.29-1
    System76 CLI tool for installing firmware updates
local/system76-firmware-daemon 1.0.29-1
    System76 systemd service that exposes a DBUS API for handling firmware updates
local/system76-io-dkms 1.0.1-1
    This DKMS driver controls System76 IO devices
local/system76-power 1.1.17-1
    System76 Power Management
```

**Issue/Bug Description:**

Every once in a while, the system76-power daemon starts seriously misbehaving. In general, this consists of the following:
- 100% CPU usage on one core, as reported by `htop`
- Running `system76-power graphics power off` immediately deadlocks
- Inability to terminate the daemon
  - It times out when you try to use `systemctl stop system76-power`
  - `kill -9 $(pidof system76-power)` has no effect
- Finally, rebooting the system results in the following message on the TTY: `failed to start system76-power daemon`. After getting this message, the entire system is locked up until I force a hard shutdown using the power button.

**Steps to reproduce (if you know):**

I don't have a 100% reliable way to reproduce this, but it seems to occur maybe 3 out of every 4 reboots on my system. If there is any other relevant information you think I can provide (journal logs, etc.) just let me know.

**Expected behavior:**

Basically none of what I listed above should be happening.

**Other Notes:**

I am running Arch Linux with a stock 5.14 kernel on an Oryx Pro.
